### PR TITLE
Autodetect mode on daemon

### PR DESCRIPTION
**Pull request by bn222**

### CODE REVIEW

File: internal/controller/dpuoperatorconfig_controller.go [hunk 1]
```diff
@@ -120,7 +120,7 @@ func (r *DpuOperatorConfigReconciler) createCommonData(cfg *configv1.DpuOperator
 	data := map[string]string{
 		"Namespace":              "openshift-dpu-operator",
 		"ImagePullPolicy":        r.imagePullPolicy,
-		"Mode":                   cfg.Spec.Mode,
+		"Mode":                   "auto",
```
Comment (wizhaoredhat) [inline internal/controller/dpuoperatorconfig_controller.go:123]: Is this intended to be hardcoded? Or left over from debug?

Comment (bn222) [reply internal/controller/dpuoperatorconfig_controller.go:123]: intented.

**Pull request closed without merging**